src/lib/protocols/alicloud.c

```diff
@@ -0,0 +1,88 @@
+/*
+ * alicloud.c
+ *
+ * Copyright (C) 2022 - ntop.org
+ *
+ * nDPI is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * nDPI is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with nDPI.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+
+#include "ndpi_protocol_ids.h"
+
+#define NDPI_CURRENT_PROTO NDPI_PROTOCOL_ALICLOUD
+
+#include "ndpi_api.h"
+
+static void ndpi_int_alicloud_add_connection(struct ndpi_detection_module_struct * const ndpi_struct,
+                                             struct ndpi_flow_struct * const flow)
+{
+  NDPI_LOG_INFO(ndpi_struct, "found alicloud\n");
+
+  if (flow->guessed_host_protocol_id == NDPI_PROTOCOL_UNKNOWN)
+  {
+    ndpi_set_detected_protocol(ndpi_struct, flow, NDPI_PROTOCOL_UNKNOWN, NDPI_PROTOCOL_ALICLOUD,
+                               NDPI_CONFIDENCE_DPI);
+  } else {
+    ndpi_set_detected_protocol(ndpi_struct, flow, flow->guessed_host_protocol_id, NDPI_PROTOCOL_ALICLOUD,
+                               NDPI_CONFIDENCE_DPI);
```

Comment:
Why this "if" block? Shoudn't we have only the second `ndpi_set_detected_protocol` call? Am I missing something?

Reply:
Copy Pasta mistake, sorry.